### PR TITLE
SEO no/index and slug urls for jobs

### PR DESCRIPTION
**Pull request by wahedF22**
SEO tasks

### CODE REVIEW

File: apps/nextapp/src/components/JobDetail/JobDescription.js [hunk 1]
```diff
@@ -28,7 +30,10 @@ const JobDescription = ({
       />
       <JobBenefits benefits={benefits} />
       <div className="h-[1px] opacity-40 bg-[#949191] w-full"></div>
-      <AboutCompany about_organization={about_organization} email={email} />
+      <AboutCompany
+        about_organization={about_organization ? about_organization : ''}
```
Comment (NurseIndia) [inline apps/nextapp/src/components/JobDetail/JobDescription.js:34]: Empty string ?

File: apps/nextapp/src/components/Layout.js [hunk 2]
```diff
@@ -26,10 +33,13 @@ export default function Layout({ children, employer, title = 'Nurse India Organi
     }
   }, []);
 
+  let noIndexFlag = noIndexTitleList.indexOf(title) >= 0 ? true : false;
```
Comment (NurseIndia) [inline apps/nextapp/src/components/Layout.js:36]: individually you added no index in few pages then what is the purpose of that ?

File: apps/nextapp/src/components/Layout.js [hunk 3]
```diff
@@ -26,10 +33,13 @@ export default function Layout({ children, employer, title = 'Nurse India Organi
...
+
   return (
     <div>
       <Head>
-        <title>{title}</title>
+        <title>{title ? title : ''}</title>
```
Comment (NurseIndia) [inline apps/nextapp/src/components/Layout.js:41]: why empty string again ? pls check what will happen if you give empty string

File: packages/ui/stories/JobCard.jsx [hunk 4]
```diff
@@ -4,7 +4,7 @@ import PropTypes from 'prop-types';
 import { CloseIcon, CurrencyIcon, MedalIcon } from './assets/Logos';
 import { UserGroupIcon, MapPinIcon, CameraIcon } from '@heroicons/react/20/solid';
 import { Button } from './Button';
-import { parseHTML, toastedNote } from '../../../apps/nextapp/src/utils/Helper';
+import { hypenateString, parseHTML, toastedNote } from '../../../apps/nextapp/src/utils/Helper';
```
Comment (NurseIndia) [inline packages/ui/stories/JobCard.jsx:7]: spelling

File: packages/ui/stories/JobCard.jsx [hunk 5]
```diff
@@ -162,12 +162,17 @@ const JobCard = ({
   const [modalData, setModalData] = useState([]);
 
   const descData = parseHTML(jobDescription);
+
+  const constructURL = `${hypenateString(jobTitle)}-${hypenateString(author)}-${hypenateString(
+    jobLocation
+  )}-`;
```
Comment (NurseIndia) [inline packages/ui/stories/JobCard.jsx:168]: why there is a - at the end instead we can put it down

**Review by NurseIndia — CHANGES_REQUESTED** [review]

File: apps/nextapp/pages/about.js [hunk 6]
```diff
@@ -3,19 +3,21 @@ import ArticleSection from '../src/components/Home/ArticleSection';
 import CtaSection from '../src/components/Home/CtaSection';
 import FeaturedCompany from '../src/components/Home/FeaturedCompany';
 import HeroSection from '../src/components/Home/HeroSection';
-import JobsByLocation from '../src/components/Home/JobsByLocation';
 import PricingSection from '../src/components/Home/PricingSection';
 import SubscribeUpdates from '../src/components/Home/SubscribeUpdates';
 import Layout from '../src/components/Layout';
+import { titleDescription } from '../src/data';
```
[the remark this answers is not in the document]
Comment (wahedF22) [reply apps/nextapp/pages/about.js:9]: changed to documentMetadata

File: apps/nextapp/pages/corporateclients.js [hunk 7]
```diff
@@ -1,85 +1,91 @@
...
-                    style={{ boxShadow: 'inset 0px 0px 54px rgba(56, 59, 147, 0.12);' }}>
-                    <div className=" w-full flex my-auto justify-center">
-                      <img src={image} alt="company_image" />
+                    key={idx}
+                    className="p-2 w-[165px] h-40 md:w-48 md:h-48 mx-auto bg-white border-gray-200 border-[0.5px] rounded-lg z-10">
+                    <div
+                      className="h-full flex justify-center px-4 rounded-lg border-gray-100 border-[0.5px]"
+                      style={{ boxShadow: 'inset 0px 0px 54px rgba(56, 59, 147, 0.12);' }}>
+                      <div className=" w-full flex my-auto justify-center">
+                        <img src={image} alt="company_image" />
```
[the remark this answers is not in the document]
Comment (wahedF22) [reply apps/nextapp/pages/corporateclients.js:46]: not my change. Let me know if you still want me to change

File: apps/nextapp/pages/corporateclients.js [hunk 8]
```diff
@@ -1,85 +1,91 @@
...
-                    <div className="my-auto">
-                      <img src={image} alt="company_image" />
+                    key={idx}
+                    style={{ boxShadow: '0px 14px 24px rgba(0, 0, 0, 0.05)' }}
+                    className="p-5 w-[165px] h-40 lg:w-[18rem] lg:h-[15rem] xl:w-[23.75rem] xl:h-[17.5rem] mx-auto bg-white border-gray-200 border-[0.5px] rounded-xl z-10">
+                    <div
+                      className="w-full h-full flex justify-center rounded-lg border-gray-100 border-[0.5px]"
+                      style={{ boxShadow: 'inset 0px 0px 54px rgba(56, 59, 147, 0.12);' }}>
+                      <div className="my-auto">
+                        <img src={image} alt="company_image" />
```
[the remark this answers is not in the document]
Comment (wahedF22) [reply apps/nextapp/pages/corporateclients.js:64]: not my change. Let me know if you still want me to change

File: apps/nextapp/pages/jobs/[id].js [hunk 9]
```diff
@@ -12,30 +12,39 @@ import { useRelatedJobs } from '../../src/lib/services/dropDownServices';
...
 export async function getServerSideProps(context) {
-  const { id } = context?.query;
+  let query = context?.query?.id;
+  let queryArr = query.split('-');
+  let id = queryArr[queryArr.length - 1];
+
   const res = await fetch(`https://api.nurseindia.org/v1/job_postings/${id}`);
   const jobDetails = await res.json();
+
+  const resRelatedJobs = await fetch(
```
[the remark this answers is not in the document]
Comment (wahedF22) [reply apps/nextapp/pages/jobs/[id].js:27]: changed to responseRelatedJobs

File: apps/nextapp/pages/jobs/[id].js [hunk 10]
```diff
@@ -12,30 +12,39 @@ import { useRelatedJobs } from '../../src/lib/services/dropDownServices';
...
+    `https://api.nurseindia.org/v1/job_postings/${id}/related_jobs`
+  );
+  const relatedJobs = await resRelatedJobs.json();
```
[the remark this answers is not in the document]
Comment (wahedF22) [reply apps/nextapp/pages/jobs/[id].js:30]: changed to responseRelatedJobs

File: apps/nextapp/src/components/JobDetail/JobDescription.js (hunk 1 above)
Comment (wahedF22) [reply apps/nextapp/src/components/JobDetail/JobDescription.js:34]: In signed in scenario this line was giving error. so had to give some string to it.

File: apps/nextapp/src/components/Layout.js (hunk 2 above)
Comment (wahedF22) [reply apps/nextapp/src/components/Layout.js:36]: Had to manually use the noindex meta tag in Pages which used ui/stories/Layout component. All other pages which used src/components/Layout in those used this flag.

File: packages/ui/stories/JobCard.jsx (hunk 4 above)
Comment (wahedF22) [reply packages/ui/stories/JobCard.jsx:7]: corrected the spelling

**Pull request merged**